### PR TITLE
Test up to Sage 10.1

### PR DESCRIPTION
**Pull request by fchapoton**
to test 10.1 and 10.2

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,13 +15,13 @@ jobs:
       fail-fast: false
       matrix:
         sage-version:
-          - '9.2'
-          - '9.3'
```
Comment (rparini) [inline .github/workflows/test.yml:19]: We may want to keep these but I'm not sure how far back we should be testing. Do you know if Sage has an official deprecation/support policy for older versions?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -15,13 +15,13 @@ jobs:
...
           - '9.4'
           - '9.5'
           - '9.6'
           - '9.7'
           - '9.8'
+          - '10.1'
+          - '10.2'
```
Comment (rparini) [inline .github/workflows/test.yml:24]: Worth testing 10.0 too?

**Review by rparini — CHANGES_REQUESTED** [review]
- There's some type coersion error in 10.1 that would need to be resolved https://github.com/abelfunctions/abelfunctions/actions/runs/7331719628/job/19964783266?pr=221
- The 10.2 compilation error looks due to https://github.com/abelfunctions/abelfunctions/issues/214
- I'll have a look at the [test-pip-install](https://github.com/abelfunctions/abelfunctions/actions/runs/7331719628/job/19964782707?pr=221#logs) failure. It might be that it doesn't work on forked PRs

File: .github/workflows/test.yml (hunk 2 above)
Comment (fchapoton) [reply .github/workflows/test.yml:24]: I think there is no docker for sage 10.0. But maybe you use conda instead ?

Comment (rparini) [reply .github/workflows/test.yml:24]: Yeah currently the workflow uses conda

**Review by rparini — APPROVED** [review]

File: abelfunctions/riemann_surface_path_factory.py [hunk 3]
```diff
@@ -155,15 +155,16 @@ def __init__(
...
+        f = self.riemann_surface.f
         if not base_sheets:
-            x, y = self.riemann_surface.f.parent().gens()
-            p = self.riemann_surface.f(self.base_point, y).univariate_polynomial()
+            x, y = f.parent().gens()
+            p = f.change_ring(CC)(self.base_point, y).univariate_polynomial()
             roots = p.roots(CDF, multiplicities=False)
             base_sheets = array(roots, dtype=complex)
         else:
             f = self.riemann_surface.f
```
Comment (slel) [inline abelfunctions/riemann_surface_path_factory.py:165]: Line 165 could be removed, now that line 158 is in.

**Pull request merged**